### PR TITLE
Li 41 user chapters and chart points endpoints

### PR DESCRIPTION
**Pull request by Gujansko**

### CODE REVIEW

**Review by Misiek22113 — APPROVED** [review]

File: src/services/UserService.ts [hunk 1]
```diff
@@ -39,6 +45,68 @@ class UserService {
...
+                const chapterQuizzesIds = chapterQuizzes.map(({ _id }) => _id.toString());
+
+                const isChapterDone = chapterDoneIds.includes(chapterId.toString());
+                const doneQuizzes = chapterQuizzesIds.filter((chapterQuizId) => quizDoneIds.includes(chapterQuizId));
+
+                return {
+                    chapterId,
+                    maximumQuizzes: chapterQuizzesIds.length,
+                    doneQuizzes: isChapterDone ? chapterQuizzesIds.length : doneQuizzes.length,
+                    percentage: isChapterDone ? 100 : Math.floor(doneQuizzes.length / chapterQuizzesIds.length) * 100,
```
[the remark this answers is not in the document]
Comment (olekgolus11) [reply src/services/UserService.ts:72]: like this

```ts
const doneQuizzes = chapterQuizzesIds.filter((chapterQuizId) => quizDoneIds.includes(chapterQuizId));

                return {
                    chapterId,
                    maximumQuizzes: chapterQuizzesIds.length,
                    doneQuizzes: doneQuizzes.length,
                    percentage: Math.floor(doneQuizzes.length / chapterQuizzesIds.length) * 100,
                    }
```

File: src/services/UserService.ts [hunk 2]
```diff
@@ -39,6 +45,68 @@ class UserService {
 
         await UserModel.create(userData);
     }
+
+    async getUserChapters(authToken: string): Promise<object[]> {
+        const userId = getUserIdFromToken(authToken);
```
Comment (olekgolus11) [inline src/services/UserService.ts:50]: nice type `object[]` :D 🗡️

File: src/services/UserService.ts [hunk 3]
```diff
@@ -39,6 +45,68 @@ class UserService {
...
+                    doneQuizzes: isChapterDone ? chapterQuizzesIds.length : doneQuizzes.length,
+                    percentage: isChapterDone ? 100 : Math.floor(doneQuizzes.length / chapterQuizzesIds.length) * 100,
+                };
+            }),
+        );
+        return chaptersWithProgress;
+    }
+
+    async getUserActivity(authToken: string, startDate: Date, duration: number): Promise<object[]> {
+        const userId = getUserIdFromToken(authToken);
```
Comment (olekgolus11) [inline src/services/UserService.ts:80]: nice type `object[]` :D 🗡️

**Review by olekgolus11 — COMMENTED** [review]
good shit

**Pull request merged**